apps/signing/tests/test_views.py

```diff
@@ -53,10 +57,18 @@ def test_not_authenticated(self):
         response = self.client.put(self.url(self.guid, '12.5'))
         assert response.status_code == 401
 
-    def test_addon_does_not_exist(self):
-        response = self.put(self.url('foo', '12.5'))
-        assert response.status_code == 404
-        assert response.data['error'] == 'Could not find addon.'
+    @mock.patch('devhub.views.auto_sign_version')
+    def test_addon_does_not_exist(self, sign_version):
+        guid = '@create-version'
+        qs = Addon.unfiltered.filter(guid=guid)
+        assert not qs.exists()
+        response = self.put(addon=guid, version='1.0')
+        assert response.status_code == 201
+        assert qs.exists()
+        addon = qs.get()
+        assert addon.has_author(self.user)
+        assert not addon.is_listed
```

Comment:
Good call. The add-on's status is not being set correctly but the version is. This will take some debugging.